classes/migration/upgrade/v3_4_0/I6895_Institutions.inc.php

```diff
@@ -0,0 +1,100 @@
+    {
+        $institutionsMigrations = new InstitutionsMigration($this->_installer, $this->_attributes);
+        $institutionsMigrations->up();
+
+        // Requires that institution tables are already there
+        if (Schema::hasTable('institutional_subscriptions') && Schema::hasTable('institutions') && Schema::hasTable('institution_settings') && Schema::hasTable('institution_ip')) {
+            if (!Schema::hasColumn('institutional_subscriptions', 'institution_id')) {
+
+                // Add the new column institution_id to the table insitutional_substriptions
+                Schema::table('institutional_subscriptions', function (Blueprint $table) {
+                    $table->bigInteger('institution_id');
+                });
+
+                // pkp/pkp-lib#6895 Migrate all institutions form institutional subscriptions into new databases
+                if (Schema::hasColumn('institutional_subscriptions', 'institution_name') && Schema::hasTable('institutional_subscription_ip')) {
+                    $institutionalSubscriptions = DB::table('institutional_subscriptions AS i')
+                        ->select('i.institutional_subscription_id', 'i.subscription_id', 'i.institution_name', 's.journal_id', 'j.primary_locale')
+                        ->join('subscriptions AS s', 's.subscription_id', '=', 'i.subscription_id')
+                        ->join('journals AS j', 'j.journal_id', '=', 's.journal_id')
+                        ->get();
+
+                    foreach ($institutionalSubscriptions as $institutionalSubscription) {
+                        $institutionId = DB::table('institutions')->insertGetId(['context_id' => $institutionalSubscription->journal_id]);
+                        if ($institutionId) {
+                            DB::table('institution_settings')->insert(['institution_id' => $institutionId, 'setting_name' => 'name', 'setting_value' => $institutionalSubscription->institution_name, 'locale' => $institutionalSubscription->primary_locale]);
+
+                            $affected = DB::table('institutional_subscriptions')
+                                ->where('institutional_subscription_id', $institutionalSubscription->institutional_subscription_id)
+                                ->update(['institution_id' => $institutionId]);
+
+                            // Get IP ranges
+                            $ipRanges = DB::table('institutional_subscription_ip')
+                                ->select('ip_string', 'ip_start', 'ip_end')
+                                ->where('subscription_id', '=', $institutionalSubscription->subscription_id)
+                                ->get();
+                            foreach ($ipRanges as $ipRange) {
+                                DB::table('institution_ip')->insert(['institution_id' => $institutionId, 'ip_string' => $ipRange->ip_string, 'ip_start' => $ipRange->ip_start, 'ip_end' => $ipRange->ip_end]);
+                            }
```

Comment:
Because I use InstitutionsMigration to create the new tables, and because I would need to add the new column and remove the old one, also to rename a column and table... I will leave it as it is and see what Alec says...

Reply:
Either approach would be OK with me. There won't be a lot of data here in general.

Reply:
OK, I will leave it as it is... :+1: